sdk/src/client/api/block_builder/input_selection/remainder.rs

```diff
@@ -118,30 +123,82 @@ impl InputSelection {
 
         let native_tokens_diff = get_native_tokens_diff(&input_native_tokens, &output_native_tokens)?;
 
-        if inputs_sum == outputs_sum && native_tokens_diff.is_none() {
+        let mut input_mana = 0;
+
+        for input in &self.selected_inputs {
+            input_mana += input.output.available_mana(
+                &self.protocol_parameters,
+                input.output_id().transaction_id().slot_index(),
+                self.slot_index,
+            )?;
+            // TODO rewards https://github.com/iotaledger/iota-sdk/issues/1310
+        }
+
+        let output_mana = self.outputs.iter().map(|o| o.mana()).sum::<u64>() + self.mana_allotments;
+
+        if input_amount == output_amount && input_mana == output_mana && native_tokens_diff.is_none() {
             log::debug!("No remainder required");
             return Ok((None, storage_deposit_returns));
         }
 
+        let amount_diff = input_amount
+            .checked_sub(output_amount)
+            .ok_or(BlockError::ConsumedAmountOverflow)?;
+        let mana_diff = input_mana
+            .checked_sub(output_mana)
+            .ok_or(BlockError::ConsumedManaOverflow)?;
+
+        // If there is only a mana remainder, try to fit it in an automatically transitioned output.
+        if input_amount == output_amount && input_mana != output_mana && native_tokens_diff.is_none() {
+            let filter = |output: &Output| {
+                output
+                    .chain_id()
+                    .as_ref()
+                    .map(|chain_id| self.automatically_transitioned.contains(chain_id))
+                    .unwrap_or(false)
+                    // Foundries can't hold mana so they are not considered here.
+                    && !output.is_foundry()
```

Comment:
should this also have delegation ignored?

Reply:
No, they can't be automatically transitioned